packages/ui-scripts/lib/publish.js

```diff
@@ -21,61 +21,113 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-const { getPackageJSON } = require('@instructure/pkg-utils')
-const { error, info } = require('@instructure/command-utils')
+const { getPackageJSON, getPackages } = require('@instructure/pkg-utils')
+const {
+  error,
+  info,
+  runCommandSync,
+  runCommandAsync,
+  confirm
+} = require('@instructure/command-utils')
 
-const { publishPackages, createNPMRCFile } = require('./utils/npm')
-const { isReleaseCommit, setupGit } = require('./utils/git')
 const { getConfig } = require('./utils/config')
+const {
+  checkWorkingDirectory,
+  isReleaseCommit,
+  setupGit
+} = require('./utils/git')
+const { createNPMRCFile } = require('./utils/npm')
 
 try {
   const pkgJSON = getPackageJSON()
   // Arguments
-  // 1: publish type. defaults to latest. If set to 'maintenance', it will publish with vx_maintenance tag
-  // e.g.: ui-scripts --publish maintenance
-  publish({
+  // 1: version to publish. If current version, use 'current' or do not pass any arguments otherwise e.g.: 8.1.3
+  // 2: publish type. defaults to current. If set to 'maintenance', it will publish with vx_maintenance tag
+  // e.g.: ui-scripts --publish 5.12.2 maintenance
+  const releaseVersion =
+    process.argv[3] === 'current' || !process.argv[3]
+      ? pkgJSON.version
+      : process.argv[3]
+  fixPublish({
     packageName: pkgJSON.name,
     currentVersion: pkgJSON.version,
-    packageConfig: getConfig(pkgJSON),
-    releaseType: process.argv[3]
+    releaseVersion: releaseVersion,
+    config: getConfig(pkgJSON),
+    releaseType: process.argv[4]
   })
 } catch (err) {
   error(err)
   process.exit(1)
 }
 
-async function publish(options) {
-  const {
-    packageName,
-    currentVersion,
-    packageConfig,
-    releaseType = 'latest'
-  } = options
-
+async function fixPublish({
```

Comment:
Probably copy-paste, but now it should be called "publish"

Reply:
Done